## Preliminary EDA for 2018 dataset

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [3]:
pd.set_option('display.max_columns',None)

### Import Bridge data

*OJO TAMBIEN HABRÍA QUE SUBIR LOS DATOS DE ELEVATIONS O DE PRECIPITACIONES SI METEMOS ESTADOS CON OTRO CLIMA*

We are creating a dataframe from the zip file obtained from FHWA dataset for all the bridges in the US National Bridge Inventory (NBIs) on 2018.

In [4]:
total18 = pd.read_csv('2018hwybronefiledel.zip',compression = 'zip',sep = ',',encoding='latin-1')

/home/dsc/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3020: DtypeWarning: Columns (1,5,7,17,31,36,37,38,39,71,73,75,96,98,102,107,108,109,123,124,135) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [5]:
total18.head()

STATE_CODE_001 STRUCTURE_NUMBER_008  RECORD_TYPE_005A  ROUTE_PREFIX_005B  \
0               1      00000000000S702                 1                6.0   
1               1      00000000000S703                 1                6.0   
2               1      0000000000M0022                 1                8.0   
3               1      000000883039900                 1                4.0   
4               1      000001014002450                 1                3.0   

   SERVICE_LEVEL_005C ROUTE_NUMBER_005D  DIRECTION_005E HIGHWAY_DISTRICT_002  \
0                 1.0             04007             0.0                    9   
1                 1.0             04003             0.0                    9   
2                 1.0             00000             0.0                    0   
3                 1.0             00088             0.0                    2   
4                 1.0             00101             0.0                    2   

   COUNTY_CODE_003  PLACE_CODE_004          FEATURES_DESC_006A  \
0             53.0             0.0  '           PERDIDO CREEK'   
1             53.0          3004.0  '           PERDIDO CREEK'   
2            113.0         27568.0  'UCHEE CREEK             '   
3             59.0             0.0  'LITTLE BEAR CR. DAM SPW '   
4             79.0             0.0  'TENNESSEE RIVER         '   

   CRITICAL_FACILITY_006B  FACILITY_CARRIED_007                 LOCATION_009  \
0                     NaN  '  IRR BIA RTE 4007'  '  11.4KM NW OF ATMORE  AL'   
1                     NaN  '  IRR BIA RTE 4003'  '       ON LYNN MCGEE ROAD'   
2                     NaN  '101ST AIRBORNE RD '  '3.4 KM S OF SUNSHINE RD  '   
3                     NaN  'CO. RD 88         '  'LITTLE BEAR CR DAM       '   
4                     NaN  'SR 101            '  'ON WHEELER DAM           '   

   MIN_VERT_CLR_010  KILOPOINT_011  BASE_HWY_NETWORK_012 LRS_INV_ROUTE_013A  \
0             99.99          1.500                   0.0         0000000000   
1             99.99          1.100                   0.0         0000000000   
2             99.99          0.000                   0.0                NaN   
3             99.99          0.000                   0.0                NaN   
4              4.52         39.429                   0.0                NaN   

   SUBROUTE_NO_013B     LAT_016    LONG_017  DETOUR_KILOS_019  TOLL_020  \
0               0.0  31061110.0  87341340.0                 3         3   
1               0.0  31062020.0  87340890.0                 3         3   
2               NaN  32174328.0  84583800.0                18         3   
3               NaN  34270600.0  87581200.0                16         3   
4               NaN  34481800.0  87225400.0                42         3   

   MAINTENANCE_021  OWNER_022  FUNCTIONAL_CLASS_026  YEAR_BUILT_027  \
0               62         62                     9          1999.0   
1               62         62                     9          2002.0   
2               74         74                     9          1942.0   
3               67         67                     8          1974.0   
4               67         67                     6          1937.0   

   TRAFFIC_LANES_ON_028A  TRAFFIC_LANES_UND_028B  ADT_029  YEAR_ADT_030  \
0                      2                       0     50.0        2014.0   
1                      2                       0    159.0        2017.0   
2                      2                       0    375.0        2017.0   
3                      2                       0    300.0        2013.0   
4                      2                       0   6200.0        2014.0   

  DESIGN_LOAD_031  APPR_WIDTH_MT_032  MEDIAN_CODE_033  DEGREES_SKEW_034  \
0               5                9.7                0                30   
1               5                6.5                0                 0   
2               4               10.4                0                 0   
3               5               11.6                0            

In [6]:
total18.shape

(616096, 137)

There are 616096 bridges on the datasets and 137 columns with variables.


In [7]:
total18.columns

Index(['STATE_CODE_001', 'STRUCTURE_NUMBER_008', 'RECORD_TYPE_005A',
       'ROUTE_PREFIX_005B', 'SERVICE_LEVEL_005C', 'ROUTE_NUMBER_005D',
       'DIRECTION_005E', 'HIGHWAY_DISTRICT_002', 'COUNTY_CODE_003',
       'PLACE_CODE_004',
       ...
       'DTL_TYPE_OF_IMP', 'SPECIAL_CODE', 'STEP_CODE', 'STATUS_WITH_10YR_RULE',
       'SUFFICIENCY_ASTERC', 'SUFFICIENCY_RATING', 'STATUS_NO_10YR_RULE',
       'CAT10', 'CAT23', 'CAT29'],
      dtype='object', length=137)

In [14]:
total18.dtypes

STATE_CODE_001              int64
STRUCTURE_NUMBER_008       object
RECORD_TYPE_005A            int64
ROUTE_PREFIX_005B         float64
SERVICE_LEVEL_005C        float64
ROUTE_NUMBER_005D          object
DIRECTION_005E            float64
HIGHWAY_DISTRICT_002       object
COUNTY_CODE_003           float64
PLACE_CODE_004            float64
FEATURES_DESC_006A         object
CRITICAL_FACILITY_006B    float64
FACILITY_CARRIED_007       object
LOCATION_009               object
MIN_VERT_CLR_010          float64
KILOPOINT_011             float64
BASE_HWY_NETWORK_012      float64
LRS_INV_ROUTE_013A         object
SUBROUTE_NO_013B          float64
LAT_016                   float64
LONG_017                  float64
DETOUR_KILOS_019            int64
TOLL_020                    int64
MAINTENANCE_021             int64
OWNER_022                   int64
FUNCTIONAL_CLASS_026        int64
YEAR_BUILT_027            float64
TRAFFIC_LANES_ON_028A       int64
TRAFFIC_LANES_UND_028B      int64
ADT_029       

As a first approach, we will be interested on working with the bridges of Pennsylvania and Ohio, whose codes are 42 and 39 respectively.

In [15]:
mystates = [42,39]
df18 = total18[total18['STATE_CODE_001'].isin(mystates)]

In [16]:
df18.shape

(50014, 137)

In [17]:
# Let's save this dataframe as a csv file since this will be the dataframe we will be performing our EDA on.
df18.to_csv (r'./df18.csv', index = None, header=True)

In [18]:
df18 = pd.read_csv('df18.csv')

/home/dsc/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3020: DtypeWarning: Columns (1,17,75,98,102,123,124,135) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


### Clear and Merge Data

We will perform several filters in order to clear our data and get some useful insights. 
The filters will be developed from generic to more specific features.

We will also need to add columns and transform some variables.

Our filters and merge operations will be: 


1) Selecting only bridges: the FHWA inventory includes data on both bridges and culverts
2) Limit the dataset to bridges built after 1900: bridges built before ASTM standards (founded in 1898) would have had inconsistent standards for material acceptance in the bridge

In [22]:
#The culverts will be dropped out of our dataset

data18 = df18.loc[df18['CULVERT_COND_062']=='N']

In [23]:
data18.shape

(45552, 137)

In [24]:
#Drop bridges built before 1900 due to material standards

data18 = data18[data18['YEAR_BUILT_027']>1900]

In [25]:
data18.shape

(43993, 137)